### DISCUSSION TITLE
set a useState from within p5 :)

### DISCUSSION THREAD

**Discussion by LasseMunk** [P5-wrapper/react#306] [General]
Hi I am following the instruction to pass a useState's state as a prop using the sketch type in typescript. 

I would like my state to be outside the p5 component, and be shared among several p5 instances.

However I need a p5 instance to be able to change my useState's state -> utilitizing the set function provided by useState. 

I am unsure on how to pass that set function to the p5 instance. 

I am currently trying this:

```typescript
.....
type MySketchProps = SketchProps & {
	amplitudeHistory: number[];
	setAmplitudeHistory: (amplitudeHistory: number[]) => void;
};

.....
export const ArcAngles2DSummingMultiple: Sketch<MySketchProps> = (p5) => {
	let amplitudeHistory: number[] = [];
	let setAmplitudeHistory: (amplitudeHistory: number[]) => void;

	p5.setup = () => {
		p5.createCanvas(windowWidth, windowHeight);
		p5.textFont(P5_TEXT_FONT);
	};

	p5.updateWithProps = (props) => {
		if (props.amplitudeHistory) amplitudeHistory = props.amplitudeHistory;
		if (props.setAmplitudeHistory)
			setAmplitudeHistory = props.setAmplitudeHistory;
	};
...
```

**Comment by jamesrweb** [comment]
Please show me your draw function, usage of the prop and let variable and also how you're passing it in, etc in a minimal example I can reproduce.
